flycheck.el

```diff
@@ -11916,16 +11920,51 @@ See URL `https://github.com/brigade/scss-lint'."
                     '(bold error)
                   'success)))))))
 
+(defun flycheck-stylelint-config-exists-p (checker)
+  "Whether there is a valid stylelint config for the current buffer."
+  (eql 0 (flycheck-call-checker-process
+          checker nil nil nil
+          "--print-config" (or buffer-file-name "index.js"))))
+
+(defun flycheck-stylelint-verify (checker)
+  "Verify stylelint setup for CHECKER."
+  (let ((have-config (flycheck-stylelint-config-exists-p checker)))
+    (list
+     (flycheck-verification-result-new
+      :label "configuration available"
+      :message (if have-config "yes" "no config file found")
+      :face (if have-config 'success '(bold error)))
+     (flycheck-verification-result-new
+      :label "stylecheck version"
+      :message (number-to-string (flycheck-stylelint-get-major-version checker))
+      :face 'success))))
+
+(defun flycheck-stylelint-get-major-version (checker)
+  "Return major version of stylelint."
+  (let ((output-buffer (generate-new-buffer " *temp*")))
+    (unwind-protect
+        (progn
```

Comment:
Could we use `with-temp-buffer` for this?

Reply:
I did it this way because `flycheck-call-checker-process` needs to run in context of the current ("flychecked") buffer, but we want to put the output to a different buffer.  With temp buffer I can rewrite it as this

```elisp
(let ((cb (current-buffer)))
  (with-temp-buffer
    (let ((temp-buffer (current-buffer)))
      (with-current-buffer cb
        (flycheck-call-checker-process
         checker nil temp-buffer nil "--version"))
      (string-to-number (car (split-string (buffer-string) "\\."))))))
```

Maybe it's a bit cleaner?